saleor/core/payments.py

```diff
@@ -0,0 +1,60 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import TYPE_CHECKING, List
+
+if TYPE_CHECKING:
+    from saleor.payment.interface import PaymentData, GatewayResponse, CustomerSource
+
+
+class Gateway(Enum):
+    """Possible gateway values.
+
+    TODO: Create this in runtime based on available plugins
+    """
+
+    DUMMY = "dummy"
+    BRAINTREE = "braintree"
+    RAZORPAY = "razorpay"
+    STRIPE = "stripe"
+
+
+class PaymentInterface(ABC):
```

Comment:
I wonder if we shouldn't have it called `Abstract` or `Base` instead of `Interface`? If we ever have a need of a non-abstract method.

Reply:
This is not a _base_ for Payment, it serves as an interface for payment business logic.